src/cli.rs

```diff
@@ -152,21 +152,22 @@ pub struct Cli {
     #[structopt(skip)]
     pub is_session_read_only: bool,
 
-    // Currently deprecated in favor of --bearer, un-hide if new auth types are introduced
     /// Specify the auth mechanism.
-    #[structopt(short = "A", long, possible_values = &AuthType::variants(),
-                default_value = "basic", case_insensitive = true, hidden = true)]
-    pub auth_type: AuthType,
+    #[structopt(short = "A", long, possible_values = &AuthType::variants(), case_insensitive = true)]
+    pub auth_type: Option<AuthType>,
 
-    /// Authenticate as USER with PASS. PASS will be prompted if missing.
+    /// Authenticate as USER with PASS or with token.
     ///
-    /// Use a trailing colon (i.e. `USER:`) to authenticate with just a username.
+    /// PASS will be prompted if missing. Use a trailing colon (i.e. `USER:`)
+    /// to authenticate with just a username.
+    ///
+    /// if --auth-type=bearer then --auth expects a token
     /// {n}{n}{n}
-    #[structopt(short = "a", long, value_name = "USER[:PASS]")]
+    #[structopt(short = "a", long, value_name = "USER[:PASS] | token")]
```

Comment:
I'd capitalize `TOKEN` here, and write the last line as ``TOKEN is expected if `--auth-type=bearer`.``.

Reply:
Done